packages/framework/src/Models/NavItem.php

```diff
@@ -23,29 +25,49 @@ class NavItem
     public bool $hidden;
 
     /**
-     * @param  \Hyde\Framework\Modules\Routing\RouteContract  $route
+     * Create a new navigation menu item.
+     *
+     * @param  \Hyde\Framework\Modules\Routing\RouteContract|null  $route
      * @param  string  $title
      * @param  int  $priority
      * @param  bool  $hidden
      */
-    public function __construct(RouteContract $route, string $title, int $priority = 500, bool $hidden = false)
+    public function __construct(?RouteContract $route, string $title, int $priority = 500, bool $hidden = false)
     {
-        $this->route = $route;
+        if ($route !== null) {
+            $this->route = $route;
+        }
+
         $this->title = $title;
         $this->priority = $priority;
         $this->hidden = $hidden;
     }
 
-    public static function fromRoute(RouteContract $route): static
+    /**
+     * Create a new navigation menu item from a route.
+     */
+    public static function toRoute(RouteContract|string $route, ?string $title = null, ?int $priority = null, ?bool $hidden = null): static
```

Comment:
I don't see a use case where this would be needed. All top level routes are automatically discovered, and there are now (better and other) ways to change priorities and labels. The only time it'd be used is when adding a link to a non-root, non-documentation-index page. And that edge case I don't think warrants the added complexity. Plus, that could easily be solved by passing an href, or by creating a fake page that redirects to a nested page.

Reply:
Resolved in #54004e60f

Reply:
Note that Hyde will still accept routes being set in the config, but you won't be able to set titles and priorities, and it's not "officially" supported in the terms it's something we at this stage will make an effort to keep working and have tests for.